docs/migration/cosmoshub-3.md

```diff
@@ -0,0 +1,189 @@
+## Upgrade Procedure
+
+__Note__: It is assumed you are currently operating a full-node running gaia v2.0.14 with v0.37.15 of the _Cosmos SDK_.
+
+- The version/commit hash of Gaia v2.0.14: `86343f24f39dbc800922e5fb296b77b9d30ebaae`
+
+1. Verify you are currently running the correct version (v2.0.14) of _gaiad_:
+
+   ```bash
+    $ gaiad version --long
+    name: gaia
+    server_name: gaiad
+    client_name: gaiacli
+    version: 2.0.14
+    commit: 86343f24f39dbc800922e5fb296b77b9d30ebaae
+    build_tags: netgo,ledger
+    go: go version go1.15 darwin/amd64
+   ```
+
+2. Make sure your chain halts at the right time and date:
+    February 18, 2021 at 06:00 UTC is in UNIX seconds: `1613628000`
+
+    ```bash
+    perl -i -pe 's/^halt-time =.*/halt-time = 1613628000/' ~/.gaiad/config/app.toml
+    ```
+
+3. Export existing state from `cosmoshub-3`:
+
+   **NOTE**: It is recommended for validators and operators to take a full data snapshot at the export
+   height before proceeding in case the upgrade does not go as planned or if not enough voting power
+   comes online in a sufficient and agreed upon amount of time. In such a case, the chain will fallback
+   to continue operating `cosmoshub-3`. See [Recovery](#recovery) for details on how to proceed.
+
+   Before exporting state via the following command, the `gaiad` binary must be stopped!
+
+   ```bash
+   $ gaiad export --for-zero-height --height=<height> > cosmoshub_3_genesis_export.json
+   ```
+
+4. Verify the SHA256 of the (sorted) exported genesis file:
+
+   ```bash
+   $ jq -S -c -M '' cosmoshub_3_genesis_export.json | shasum -a 256
+   [SHA256_VALUE]  cosmoshub_3_genesis_export.json
+   ```
+
+5. At this point you now have a valid exported genesis state! All further steps now require
+v4.0.0 of [Gaia](https://github.com/cosmos/gaia). 
+Cross check your genesis hash with other peers (other validators) in the chat rooms.
+
+   **NOTE**: Go [1.15+](https://golang.org/dl/) is required!
+
+   ```bash
+   $ git clone https://github.com/cosmos/gaia.git && cd gaia && git checkout v4.0.0; make install
+   ```
+
+6. Verify you are currently running the correct version (v4.0.0) of the _Gaia_:
+
+- The version/commit hash of Gaia v4.0.0: `2bb04266266586468271c4ab322367acbf41188f`
+- The upgrade time as agreed upon by governance: **February 18, 2021 at 06:00 UTC**
+
+   ```bash
+    $ gaiad version --long
+    name: gaia
+    server_name: gaiad
+    version: 4.0.0
+    commit: 2bb04266266586468271c4ab322367acbf41188f
+    build_tags: netgo,ledger
+    go: go version go1.15 darwin/amd64
+    build_deps:
+    ...
+   ```
+
+7. Migrate exported state from the current v2.0.14 version to the new v4.0.0 version:
+
+   ```bash
+   $ gaiad migrate cosmoshub_3_genesis_export.json --chain-id=cosmoshub-4 --genesis-time=[PLACEHOLDER]> genesis.json
+   ```
+
+   The genesis time shall be the upgrade time of `2021-02-18T06:00:00Z` + `60` minutes with the subseconds truncated.
+
+   **2021-02-18T07:00:00Z**
+
```

Comment:
I found an instruction on how to do this in the [cosmoshub-2.md migration guide](https://github.com/cosmos/gaia/blob/2bb04266266586468271c4ab322367acbf41188f/docs/migration/cosmoshub-2.md#L164-L164), which might be helpful:
```
curl https://stargate.cosmos.network:26657/block\?height\=2902000 | jq -r '.result["block_meta"]["header"]["time"]'|xargs -0 date -v +60M  -j  -f "%Y-%m-%dT%H:%M:%S" +"%Y-%m-%dT%H:%M:%SZ"
```

Reply:
The upgrade will be a little different from last time. We have a `halt-time`, which will stop the daemon at a certain time, we don't really know which block it will be. But then, the daemon will be stopped and therefore not accessible via port `26657`.